### PR TITLE
Fixes swarmers breaking the SM's containment

### LINKED ISSUE #30299: Swarmers can destroy the SM chamber door
Ban bait.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #30299

### CODE REVIEW

**Comment by Repukan** [comment]
Boo, boo I say

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 1]
```diff
@@ -257,11 +257,11 @@
 /obj/machinery/door/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
 	var/isonshuttle = istype(get_area(src), /area/shuttle)
 	for(var/turf/T in range(1, src))
-		if(isspaceturf(T) || (!isonshuttle && (istype(T.loc, /area/shuttle) || istype(T.loc, /area/space))) || (isonshuttle && !istype(T.loc, /area/shuttle)))
+		if(isspaceturf(T) || (!isonshuttle && (istype(get_area(T), /area/shuttle) || istype(get_area(T), /area/space))) || (isonshuttle && !istype(get_area(T), /area/shuttle)))
```
Comment (Cyberboss) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: the result of get_area should be cached

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 2]
```diff
@@ -344,11 +344,11 @@
 /turf/closed/wall/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
 	var/isonshuttle = istype(loc, /area/shuttle)
 	for(var/turf/T in range(1, src))
-		if(isspaceturf(T) || (!isonshuttle && (istype(T.loc, /area/shuttle) || istype(T.loc, /area/space))) || (isonshuttle && !istype(T.loc, /area/shuttle)))
+		if(isspaceturf(T) || (!isonshuttle && (istype(get_area(T), /area/shuttle) || istype(get_area(T), /area/space))) || (isonshuttle && !istype(get_area(T), /area/shuttle)))
```
Comment (Cyberboss) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: ditto

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 3]
```diff
@@ -357,11 +357,11 @@
 /obj/structure/window/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
 	var/isonshuttle = istype(get_area(src), /area/shuttle)
 	for(var/turf/T in range(1, src))
-		if(isspaceturf(T) || (!isonshuttle && (istype(T.loc, /area/shuttle) || istype(T.loc, /area/space))) || (isonshuttle && !istype(T.loc, /area/shuttle)))
+		if(isspaceturf(T) || (!isonshuttle && (istype(get_area(T), /area/shuttle) || istype(get_area(T), /area/space))) || (isonshuttle && !istype(get_area(T), /area/shuttle)))
```
Comment (Cyberboss) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: ditto

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**